libass/ass_fontselect.c

```diff
@@ -256,6 +260,404 @@ static ASS_FontInfo *allocate_font_info(ASS_FontSelector *selector)
+            }
+            break;
+        }
+
+        bool fully_elidable = true;
+        for (j = 0; j < g->nb_values; j++) {
+            if (g->values[j].axis_index >= stat->nb_axes)
+                return false;
+
+            uint32_t tag = stat->axes[g->values[j].axis_index].tag;
+            int32_t value = g->values[j].value;
+
+            if (tag == FT_MAKE_TAG('w','g','h','t')) {
+                if (value != (400 << 16) &&
+                    value != (700 << 16)) {
+                    fully_elidable = false;
+                    break;
+                }
+            } else if (tag == FT_MAKE_TAG('i','t','a','l')) {
+                if (value != (1 << 16) &&
+                    value != (0)) {
+                    fully_elidable = false;
+                    break;
+                }
+            } else {
+                fully_elidable = false;
+                break;
+            }
+        }
+
+        if (fully_elidable)
+            g->flags |= STAT_FLAG_ELIDABLE_FAMILY;
+    }
+
+    stat->sorted_axes = calloc(stat->nb_axes, sizeof(struct Axis));
+    if (!stat->sorted_axes)
+        return false;
+
+    memcpy(stat->sorted_axes, stat->axes, stat->nb_axes * sizeof(struct Axis));
+    qsort(stat->sorted_axes, stat->nb_axes, sizeof(struct Axis), compare_axes);
+
+    return true;
+}
+
+/**
+ * \brief Free arrays in a STAT structure.
+ *
+ * \param stat structure to free
+ */
+static void free_stat(struct STAT *stat)
+{
+    free(stat->axes);
+    free(stat->sorted_axes);
+
+    if (stat->value_groups) {
+        for (size_t i = 0; i < stat->nb_value_groups; i++)
+            free(stat->value_groups[i].values);
+    }
+
+    free(stat->value_groups);
+}
+
+struct AppendedName {
+    uint16_t id;
+    FT_UInt index;
+    bool include_in_family;
+};
+
+struct VariantData {
+    bool *axis_satisfied;
+    struct AppendedName *appended_names;
+    FT_Long style_flags;
+    int weight;
+    uint16_t nb_appended_names;
+    bool all_in_family;
+};
+
+/**
+ * \brief Allocate arrays required to describe a named instance we'll insert.
+ *
+ * \param data structure to fill
+ * \param nb_axes number of axes to allocate data for
+ * \return success
+ */
+static bool alloc_variant_data(struct VariantData *data, size_t nb_axes)
+{
+    data->axis_satisfied = calloc(nb_axes, sizeof(bool));
+    data->appended_names = calloc(FFMAX(nb_axes, 1), sizeof(struct AppendedName));
+
+    return data->axis_satisfied && data->appended_names;
+}
+
+/**
+ * \brief Free arrays in a VariantData structure.
+ *
+ * \param stat structure to free
+ */
+static void free_variant_data(struct VariantData *data)
+{
+    free(data->axis_satisfied);
+    free(data->appended_names);
+}
+
+/**
+ * \brief Compute GDI-compatible configuration for a given named instance of a font.
+ *
+ * \param data structure to fill
+ * \param variant FreeType fvar data for the variant
+ * \param stat parsed OpenType STAT table
+ * \return success
```

Comment:
The doc doesn't represent the function.
It says ``\return success``, but the method actually return nothing.

Reply:
Good catch; corrected.